### PR TITLE
GPU: More optimizations to GPU Command List Processing and DMA Copy Optimizations

### PR DESCRIPTION
**Pull request by FernandoS27**
This PR aims to delegate engine calling to the DMA pusher, thus reducing spaghetti code. At the same time we are adding an optimized path for small DMA copies that avoid reading the whole texture but just tiny bits of it.

This optimizations greatly benefit Pokemon Sword & Shield on Cities and some routes.

### CODE REVIEW

**Comment by bukyapavankumar** [comment]
On pklg too?

**Comment by FearlessTobi** [comment]
@bukyapavankumar Our Github is meant for developer is discussion only. Please stop unrelated posting messages under PRs/issues. Thanks

**Comment by bukyapavankumar** [comment]
Ok .I have an issue with yuzu on clear linux I asked in discord no one gave me solution .I can create an issue on git.it was about libsndio . I compiled it and synlinked it but still asks for it

File: src/video_core/gpu.cpp [hunk 1]
```diff
@@ -347,7 +349,27 @@ void GPU::ProcessBindMethod(const MethodCall& method_call) {
...
+        dma_pusher->BindSubchannel(kepler_compute.get(), method_call.subchannel);
+        break;
+    case EngineID::MAXWELL_DMA_COPY_A:
+        dma_pusher->BindSubchannel(maxwell_dma.get(), method_call.subchannel);
+        break;
+    case EngineID::KEPLER_INLINE_TO_MEMORY_B:
+        dma_pusher->BindSubchannel(kepler_memory.get(), method_call.subchannel);
+        break;
+    default:
+        UNIMPLEMENTED_MSG("Unimplemented engine");
```
Comment (ogniK5377) [inline src/video_core/gpu.cpp:371]:
```suggestion
        UNIMPLEMENTED_MSG("Unimplemented engine {:04X}", static_cast<u32>(engine_id));
```

File: src/video_core/gpu.cpp [hunk 2]
```diff
@@ -347,7 +349,27 @@ void GPU::ProcessBindMethod(const MethodCall& method_call) {
     // Bind the current subchannel to the desired engine id.
     LOG_DEBUG(HW_GPU, "Binding subchannel {} to engine {}", method_call.subchannel,
               method_call.argument);
-    bound_engines[method_call.subchannel] = static_cast<EngineID>(method_call.argument);
+    auto engine_id = static_cast<EngineID>(method_call.argument);
```
Comment (ogniK5377) [inline src/video_core/gpu.cpp:352]:
```suggestion
    const auto engine_id = static_cast<EngineID>(method_call.argument);
```

**Pull request merged**